### ISSUE TITLE
bgworker process not starting in example

### ISSUE THREAD

**Issue by gtfierro** [zombodb/pgx#40]
Hi! 

I've been trying to get the [bgworker](https://github.com/zombodb/pgx/tree/master/pgx-examples/bgworker) example working but I don't see the expected log output when I run the example. Other examples like `arrays` work fine, so I believe I have pgx set up correctly.

Steps to reproduce:

0. Install `cargo-pgx`
1. Clone https://github.com/zombodb/pgx
2. Run `cargo pgx init`
3. From [`pgx-examples/bgworker`](https://github.com/zombodb/pgx/tree/master/pgx-examples/bgworker) run `cargo pgx run pg11` (to create the `data-11` folder)
4. Run `cargo pgx stop pg11`
5. Add `session_preload_libraries = 'bgworker.so'` to the last line of `~/.pgx/data-11/postgresql.conf`
6. Re-run `cargo pgx run pg11`
4. Observe the log output in `~/.pgx/11.log` (I've been using `tail -f`)
5. *Should* observe some output from example, e.g. `"Hello from inside the worker"`, but this never shows up.

I've put a `log!` call in `_PG_init` and that shows up, so the extension is getting built and included ok. I've also tried `CREATE EXTENSION bgworker;` inside the psql prompt that `cargo pgx run` drops you into, to no avail. (The `log!` calls in `_PG_init` actually happen twice, but I'm not sure why)

I've also tried with the latest stable rust (v1.46.0) and 1.47.0-nightly; both build, but I'm not seeing the output I'd expect.

I think it's most likely I'm missing some obvious configuration or trigger somewhere, but I'm not quite sure where to look next. Let me know if there's any other

**Comment by eeeebbbbrrrr** [comment]
ha!  good catch.  It supposed to be `shared_preload_libraries` not `session_preload_libraries`.  What a terrible type-o.

Make that change to your `postgresql.conf` and I bet you'll be back on track.  I'll update the example later this evening.

**Comment by gtfierro** [comment]
Thanks! That solved the immediate issue.

I also found that I had to create the `postgres` user manually --- whatever configuration process is run by `cargo pgx run pg11` or other tools didn't handle it.

**Comment by eeeebbbbrrrr** [comment]
Well. It’s intended to just be the user that ran cargo pgx install. 

Why did you need a user named “postgres”?

**Comment by eeeebbbbrrrr** [comment]
Is this the error you got?

```
    Stopping Postgres v11
building extension with features `pg11`
"cargo" "build" "--features" "pg11" "--no-default-features"
    Finished dev [unoptimized + debuginfo] target(s) in 0.08s

installing extension
     Copying control file to `/Users/e_ridge/.pgx/11.8/pgx-install/share/postgresql/extension/bgworker.control`
     Copying shared library to `/Users/e_ridge/.pgx/11.8/pgx-install/lib/postgresql/bgworker.so`
     Writing extension schema to `/Users/e_ridge/.pgx/11.8/pgx-install/share/postgresql/extension/bgworker--1.0.sql`
    Finished installing bgworker
    Starting Postgres v11 on port 28811
     [error] problem checking if database 'bgworker' exists: "/Users/e_ridge/.pgx/11.8/pgx-install/bin/psql" "-XqAt" "-h" "localhost" "-p" "28811" "template1" "-c" "select count(*) from pg_database where datname = \'bgworker\';"

psql: FATAL:  role "postgres" does not exist
```

I was able to cause that by setting the `PGUSER` environment variable (ie, `PGUSER=postgres cargo pgx run pg11`).

If you've got that set, that's the "problem".  I think the solution is for `cargo pgx run` not to pass through the various `PGXXX` environment variables.

I'll likely do that anyways, but I'd appreciate it if you can confirm that you do have a `PGUSER` envar.

**Comment by gtfierro** [comment]
Yes, I get the `FATAL: role "postgres" does not exist` error, but I don't have `PGUSER` set anywhere, so I'm not sure where that is coming from. Having trouble reproducing the error now, too, so feel free to close this issue and I can file a new one if I encounter the `postgres` user issue again and can consistently reproduce it

**Comment by eeeebbbbrrrr** [comment]
That's definitely weird.  I can't think of another way that `cargo pgx run` would want to use a user named `postgres`.  Nothing in `pgx` sets the user -- it assumes the the current user name (or, until 454eddf, `$PGUSER`).

I'm leaving this open as you have identified a few issues that I've already fixed.  Will close whenever the next release is cut.

If you can figure out how this happened, it'd be greatly appreciated!

**Comment by jamessewell** [comment]
I bet it's this from the example:

BackgroundWorker::connect_worker_to_spi(Some("postgres"), Some("postgres"));
I will PR to fix, it can be:

BackgroundWorker::connect_worker_to_spi(Some("postgres"), None); This will
connect as the user which initialized the cluster. Cheers, James

**Comment by eeeebbbbrrrr** [comment]
Oh sure.  I even searched the code base for “postgres” (including quotes) and didn’t see that. Duh.

**Comment by eeeebbbbrrrr** [comment]
released in 0.0.14
